api/admin/controller/settings.py

```diff
@@ -307,14 +304,16 @@ def _set_configuration_library(
         library = get_one(self._db, Library, short_name=info_copy.pop("short_name"))
         if not library:
             raise RuntimeError("Could not find the configuration library")
-        config = None
 
         # Validate first
         validated_data = protocol_class.library_settings_class()(**info_copy)
+
         # Attach the configuration
-        config = configuration.for_library(cast(int, library.id), create=True)
-        config.settings_dict = validated_data.dict()
-        return config
+        library_configuration = IntegrationLibraryConfiguration(
```

Comment:
I updated `for_library` to only return existing configurations, instead of also creating them. So it broke these lines unless you create the `IntegrationLibraryConfiguration` before.